src/lem/path.rs

```diff
@@ -111,77 +105,74 @@ impl LEMCTL {
         &self,
         path: &Path,
         // `map` keeps track of the updated names of variables
-        map: &mut HashMap<Var, Var>, // name -> path.name
-    ) -> Result<Self> {
+        map: &mut VarMap<Var>, // name -> path.name
+    ) -> Self {
         match self {
             LEMCTL::MatchTag(var, cases) => {
                 let mut new_cases = vec![];
                 for (tag, case) in cases {
-                    let new_case = case.deconflict(&path.push_tag(tag), &mut map.clone())?;
+                    let new_case = case.deconflict(&path.push_tag(tag), &mut map.clone());
                     new_cases.push((*tag, new_case));
                 }
```

Comment:
Occurs twice in this file. The vector, mutable at that, is unneeded. See also my comment about cloning the map.
```rust
               let new_cases: IndexMap<_, _> = cases
                    .iter()
                    .map(|(tag, case)| {
                        case.deconflict(&path.push_tag(tag), &mut map)
                            .map(|new_case| (*tag, new_case))
                    })
                    .collect::<Result<_, _>>()?;
```